File: backend/schemas/chat_schema.py
from marshmallow import Schema, fields

class GenericChatSchema(Schema):
    message=fields.String(required=True)
    chatHistory=fields.List(fields.Dict, required=False)
    knowledgeBase=fields.String(required=False)

class ImproveSuggestionSchema(Schema):
    name=fields.String(required=True)
    description=fields.String(required=True)
    type=fields.String(required=True)
    requirement=fields.String(required=True)
    suggestions=fields.List(fields.String, required=True)
    selectedSuggestion=fields.String(required=False)
    knowledgeBase=fields.String(required=False)

class ConverseRequirementSchema(Schema):
    name=fields.String(required=True)
    description=fields.String(required=True)
    type=fields.String(required=True)
    requirement=fields.String(required=True)
    chatHistory=fields.List(fields.Dict, required=False)
    knowledgeBase=fields.String(required=False)
    userMessage=fields.String(required=True)
    requirementAbbr=fields.String(required=True)

class ConverseUserStoryTaskSchema(Schema):
    name=fields.String(required=True)
    description=fields.String(required=True)
    type=fields.String(required=True)
    requirement=fields.String(required=True)
    chatHistory=fields.List(fields.Dict, required=False)
    knowledgeBase=fields.String(required=False)
    userMessage=fields.String(required=True)
    prd=fields.String(required=True)
    us=fields.String(required=False)

File: electron/schema/requirement/chat.schema.ts
import { bedrockConfigSchema } from '../helper/bedrock.options.schema';
import { z } from 'zod';

export const chatUpdateRequirementSchema = z.object({
  name: z.string(),
  description: z.string(),
  type: z.string(),
  requirement: z.string(),
  chatHistory: z.array(z.record(z.any())).optional(),
  knowledgeBase: z.string().optional(),
  bedrockConfig: bedrockConfigSchema.optional(), 
  userMessage: z.string(),
  requirementAbbr: z.enum(['BRD', 'PRD', 'UIR', 'NFR', 'BP']),
  brds: z.array(z.object({
    id: z.string(),
    title: z.string(),
    requirement: z.string()
  })).optional()
});

export type ChatUpdateRequirementRequest = z.infer<typeof chatUpdateRequirementSchema>;

export interface ChatUpdateRequirementResponse {
  response: string;
  blocked?: boolean;
  blockedReason?: string;
}
